packages/@expo/cli/src/prebuild/prebuildAsync.ts

```diff
@@ -182,3 +191,78 @@ export async function prebuildAsync(
     exp,
   };
 }
+
+async function checkCNG(platforms: ModPlatform[], projectRoot: string) {
+  const { resolveWorkflowAsync } = await import('../utils/workflow.js');
+  const { Log } = await import('../log.js');
+  const chalk = await import('chalk');
+
+  const nativePlatforms = platforms.filter(
+    (platform) => platform === 'ios' || platform === 'android'
+  );
+  if (nativePlatforms.length === 0) return;
+
+  const workflows: { platform: string; workflow: string }[] = [];
+
+  for (const platform of nativePlatforms) {
+    try {
+      const workflow = await resolveWorkflowAsync(projectRoot, platform);
+      workflows.push({ platform, workflow });
+    } catch (error) {
+      debug(`Could not determine workflow for ${platform}: ${error}`);
+    }
+  }
+
+  if (workflows.length === 0) return;
+
+  const uniqueWorkflows = [...new Set(workflows.map((w) => w.workflow))];
+
+  if (uniqueWorkflows.length === 1) {
+    const workflow = uniqueWorkflows[0];
+
+    if (workflow === 'managed') {
+      Log.warn(chalk.default.yellow(`Using Continuous Native Generation. ${cngDocs}\n`));
+    } else {
+      Log.warn(chalk.default.yellow(`NOT using Continuous Native Generation. ${cngDocs}\n`));
+    }
+  } else {
+    let message = '';
+    workflows.forEach(({ platform, workflow }) => {
+      if (workflow === 'managed') {
+        message += `The ${platform} project is using the Continuous Native Generation.\n`;
+      } else {
+        message += `The ${platform} project is NOT using the Continuous Native Generation.\n`;
+      }
+    });
+    message += `${cngDocs}\n`;
+    Log.warn(chalk.default.yellow(message));
+  }
+
+  const notConfiguredPlatforms = workflows
+    .filter(({ workflow }) => workflow === 'not-configured')
+    .map(({ platform }) => platform);
+
+  if (notConfiguredPlatforms.length > 0 && isInteractive()) {
+    const platformFolders = notConfiguredPlatforms.map((platform) =>
+      platform === 'ios' ? '/ios' : '/android'
+    );
+
+    const shouldAddToGitignore = await confirmAsync({
+      message: `Would you like to add ${platformFolders.join(' and ')} to .gitignore to use Continuous Native Generation? \n${cngDocs}`,
+      initial: true,
+    });
+
+    if (shouldAddToGitignore) {
+      const gitignorePath = path.join(projectRoot, '.gitignore');
+
+      if (!fs.existsSync(gitignorePath)) {
+        fs.writeFileSync(gitignorePath, '');
+      }
+
+      const gitignoreEntries = platformFolders.join('\n');
+      upsertGitIgnoreContents(gitignorePath, gitignoreEntries);
+
+      Log.log(chalk.default.green(`✓ Added ${platformFolders.join(' and ')} to .gitignore`));
```

Comment:
i don't believe we have a convention of using `✓` for success in similar scenarios but i don't mind it! the scenarios that came to mind for me were:
- `npx expo export` - we just print "Exported: dist" in green. 
- `npx expo customize` - we don't print anything, but you will see `✓`  if you select it in the prompt - otherwise we just print "Generating: babel.config.js" for example (if you run `npx expo customize`)

there's some opportunity to follow up on conventions here. @douglowder - is there a convention on EAS CLI that we should be following?

(no action needed on this comment)

Reply:
I removed the `✓` so it's the same as option `npx expo export`.